ORA-01000: maximum open cursors exceeded (DBD ERROR: OCIStmtExecute)

After installing ZabbixDBA and running it, I get ORA-01000 in the logs. The database has a high value of open_cursors, so it looks like a bug with not closing cursors correctly in ZabbixDBA.

Here's a typical error from the log:

[2017/10/09 11:23:30][ERROR] [ZDBA::DBIx::_fetch:95] failed to execute query on 'DBI:Oracle:host=uk01pora001;port=1521;sid=casprod1':
ORA-01000: maximum open cursors exceeded (DBD ERROR: OCIStmtExecute)


Anyone got any ideas?

--- a/lib/ZDBA/DBIx.pm
+++ b/lib/ZDBA/DBIx.pm
@@ -77,7 +77,7 @@ sub _fetch {
   $self->log->debug( sub { qq{fetching data from '%s' using '%s' with options:\n%s}, $self->dsn, $method, $self->dump($args) } );
 
   my $start = [gettimeofday];
-  my $options = { MaxRows => 1000, %{ $args->{options} || {} } };
+  my $options = { MaxRows => 1000, %{ $args->{options} // {} } };
 
   my $result = $self->dbh->$method(
     $args->{query},

--- a/lib/ZDBA/DBIx.pm
+++ b/lib/ZDBA/DBIx.pm
@@ -77,11 +77,11 @@ sub _fetch {
   $self->log->debug( sub { qq{fetching data from '%s' using '%s' with options:\n%s}, $self->dsn, $method, $self->dump($args) } );
 
   my $start = [gettimeofday];
+  my $options = { MaxRows => 1000, %{ $args->{options} || {} } };
 
   my $result = $self->dbh->$method(
     $args->{query},
-    $args->{options},
-    @{ $args->{bind_values} }
+    $options
   );
 
   my $end = [gettimeofday];

